api/src/main/java/org/apache/iceberg/expressions/StrictMetricsEvaluator.java

```diff
@@ -489,5 +500,32 @@ private boolean containsNaNsOnly(Integer id) {
           && valueCounts != null
           && nanCounts.get(id).equals(valueCounts.get(id));
     }
+
+    private boolean supportsEvaluate(int fieldId) {
+      Boolean evaluable = canEvaluate.get(fieldId);
+      if (evaluable != null) {
+        return evaluable;
+      }
+
+      evaluable = true;
+      // Cannot evaluate on complex types or repeated primitive types.
+      if (!schema.findType(fieldId).isPrimitiveType()) {
+        evaluable = false;
+      } else {
+        Integer parent = idToParent.get(fieldId);
+        while (parent != null) {
+          Type type = schema.findType(parent);
+          if (type.isListType() || type.isMapType()) {
+            evaluable = false;
```

Comment:
So you are skipping if list/map type, if I understand?  I think it makes sense to me, as I feel we have nested column stats.  but definitely like @rdblue @RussellSpitzer @aokolnychyi  to have a sanity check here on the overall direction.

Reply:
Yes, your understanding is correct.